src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -594,5 +594,23 @@ internal static Location FirstOrNone(this ImmutableArray<Location> items)
         {
             return items.IsEmpty ? Location.None : items[0];
         }
+
+        internal static bool SequenceEqual<TElement, TArg>(ImmutableArray<TElement> array1, ImmutableArray<TElement> array2, TArg arg, Func<TElement, TElement, TArg, bool> predicate)
```

Comment:
>ImmutableArray<TElement> array1 [](start = 59, length = 31)

Should this have a ```this``` modifier?

Reply:
For things like SequenceEqual, I prefer the style of having both things be arguments rather than an extension method. If you'd prefer to have it as an extension, reopen and I can change it.

---
In reply to: [327330140](https://github.com/dotnet/roslyn/pull/38769#discussion_r327330140) [](ancestors = 327330140)

Reply:
> For things like SequenceEqual, I prefer the style of having both things be arguments rather than an extension method.

It would be extremely hard to find this helper. It won't show in the completion list after the dot. One would have to be sure it exists and remember this type's name

---
In reply to: [327339134](https://github.com/dotnet/roslyn/pull/38769#discussion_r327339134) [](ancestors = 327339134,327330140)

Reply:
Alright, I'll change it.

---
In reply to: [327340766](https://github.com/dotnet/roslyn/pull/38769#discussion_r327340766) [](ancestors = 327340766,327339134,327330140)